modules/security/rbac/README.md

```diff
@@ -1,99 +1,227 @@
-Istio RBAC for Service to Service Communication
-===
+# Istio RBAC for Service to Service Communication
+
+Istio provides every workload with a strong identity - in Kubernetes, the pod's `ServiceAccount` is used to establish mTLS connections between services in the mesh. While establishing mTLS connections, sidecars in the mesh will validate certificates according to [SPIFFE's X.509 SVID spec](https://github.com/spiffe/spiffe/blob/master/standards/X509-SVID.md), which means that after the connection is established we have an authenticated identity of the other party. Then, Istio allows you to write access control policies using those identities to describe which services can communicate.
+
+Note that it means that you have to enforce mTLS between your apps for the Authorization Policies to work.
+
+## Setup a deny-all policy
+
+```yaml
+kubectl apply -n hipstershopv1v2 -f - <<EOF
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: deny-all
+  namespace: hipstershopv1v2
+spec:
+  {}
+EOF
+```
+
+At this moment, no traffic can go through the Mesh. If you try to browser the Hipstershop, you will see an error `RBAC: access denied` and some `error 403` in the logs.
+
+## Allow the Ingress Gateway to talk to the Frontend
+
+We set ourselfs out by applying a very restrictive policy. Now we need to clear communications we want to allow inside the mesh. As the policy is enforced at every Istio Proxy, we need two new policies: from outside to the Ingress Gateway and from the Ingress Gateway to the Frontend service. We are going to only allow `HTTP GET` requests first:
+
+```yaml
+kubectl apply -n hipstershopv1v2 -f - <<EOF
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: ingress-policy
+spec:
+  selector:
+    matchLabels:
+      app: istio-ingressgateway
+  action: ALLOW
+  rules:
+  - to:
+    - operation:
+        methods:
+        - GET
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: ingress-policy-to-frontend
+spec:
+  selector:
+    matchLabels:
+      app: frontend
+  action: ALLOW
+  rules:
+  - to:
+    - operation:
+        methods:
+        - GET
+EOF
+```
+
+If you browser the Hipstershop store now, you can access the frontend. Sadly, you are still getting an error:
+
+```yaml
+Uh, oh!
+Something has failed. Below are some details for debugging.
+
+HTTP Status: 500 Internal Server Error
+```
+
+If we look at the logs of the `frontend` service, we can see we still have a DENY: `could not retrieve currencies: rpc error: code = PermissionDenied desc = RBAC: access denied`
+
```

Comment:
It might be helpful here to look quickly at the logs on Envoy, and turn up the RBAC logs (`istioctl pc log <pod> --level rbac:debug`).

It would show exactly what policy failed, for example. Not super useful in these exact cases, but can be helpful for users trying to debug more complex policies.

Reply:
I added a lot more debug info there. Please review and comment. Maybe I put too much informations there... ?